http://bugs.mysql.com/bug.php?id=30356
-> just undef bool in any case, such that the code below will define it

--- include/my_global.h
+++ include/my_global.h
@@ -133,6 +133,7 @@
 #  if defined(__ppc__) || defined(__ppc64__)
 #    define WORDS_BIGENDIAN
 #  endif
+#  undef bool
 #endif /* defined(__APPLE__) && defined(__MACH__) */
 
 
